homeassistant/components/huum/config_flow.py

```diff
@@ -0,0 +1,87 @@
+"""Config flow for huum integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import aiohttp
+from huum.huum import Huum
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    huum_handler = Huum(
+        data[CONF_USERNAME],
+        data[CONF_PASSWORD],
+        session=async_get_clientsession(hass),
+    )
+
+    try:
+        await huum_handler.status()
+    except aiohttp.ClientError as client_error:
+        _LOGGER.error("Could not log in to Huum with given credentials")
+        raise InvalidAuth from client_error
+
+
+class HuumConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for huum."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            await validate_input(self.hass, user_input)
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception as err:  # pylint: disable=broad-except
+            _LOGGER.error(str(err))
```

Comment:
```suggestion
            _LOGGER.exception("Unknown error")
```

Reply:
Would this not throw away information about what went wrong? So if some exception that I have not forseen is thrown, we would silence it and just log "Unknown error" instead of outputting the actual error in the logs. Just outputting a string like this would make it hard to debug no?

Reply:
that's the funny thing. `LOGGER.exception` actually logs the whole exception. I also did not know this was a thing but its beautiful

Reply:
Refactored